nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -99,6 +102,13 @@ public ValidationResult validate(final String subject, final String value, final
         .required(false)
         .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
         .build();
+    static final PropertyDescriptor RESULTS_PER_FLOWFILE = new PropertyDescriptor.Builder()
+        .name("results-per-flowfile")
+        .displayName("Results Per FlowFile")
+        .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
+        .required(false)
+        .addValidator(StandardValidators.INTEGER_VALIDATOR)
```

Comment:
Quick remark, the validator you chose means that 0 is an acceptable value. It also means that if the user sets this property to 0 (or a negative integer), one single flow file will be generated with a JSON array containing all the documents. If that's on purpose, this should be indicated in the description. If it's not the desired behavior, I would suggest: ``StandardValidators.POSITIVE_INTEGER_VALIDATOR``.